OpenRA.Game/Network/Connection.cs

```diff
@@ -131,57 +136,65 @@ public void Dispose()
 
 	sealed class NetworkConnection : EchoConnection
 	{
-		TcpClient socket;
-		int clientId;
-		ConnectionState connectionState = ConnectionState.Connecting;
-		Thread t;
+		readonly TcpClient tcp;
+		readonly List<byte[]> queuedSyncPackets = new List<byte[]>();
+		volatile ConnectionState connectionState = ConnectionState.Connecting;
+		volatile int clientId = -1;
```

Comment:
Would you mind if we defered that to Next + 1?  I'd like to get this merged ASAP for a playtest, and the quickest/safest way to do that would be to reset this back to 0 for now.